test/simple_operations.js

```diff
@@ -1,15 +1,355 @@
+      );
+    } catch(error) {
+      assert(error.message.includes("revert"));
+    }
+
+    let new_release = await operations.release("parity", release);
+    assert.deepEqual(
+      new_release.map(v => v.valueOf()),
+      ["0", "0", "0", false],
+    );
+
+    // we successfully add a release of the parity client
+    await operations.addRelease(release, forkBlock, track, semver, critical);
+
+    // the new release should be returned by the getter
+    new_release = await operations.release("parity", release);
+    assert.deepEqual(
+      new_release.map(v => v.valueOf()),
+      [forkBlock, track, semver, critical],
+    );
+
+    // it should be set has the latest release for its track
+    new_release = await operations.latestInTrack("parity", track);
+    assert.equal(new_release, release);
+    assert(await operations.isLatest("parity", release));
+
+    // we can get the track for this release
+    const new_track = await operations.track("parity", release);
+    assert.equal(new_track, track);
+
+    // it should emit a `ReleaseAdded` event
+    const events = await watcher.get();
+
+    assert.equal(events.length, 1);
+    assert.equal(web3.toUtf8(events[0].args.client), "parity");
+    assert.equal(events[0].args.forkBlock, forkBlock);
+    assert.equal(events[0].args.release, release);
+    assert.equal(events[0].args.track, track);
+    assert.equal(events[0].args.semver, semver);
+    assert.equal(events[0].args.critical, critical);
+  });
+
+  step("should allow the owner of a client to add a checksum", async () => {
+    const operations = await SimpleOperations.deployed();
+    const watcher = operations.ChecksumAdded();
+
+    const release = "0x1234560000000000000000000000000000000000000000000000000000000000";
+    const platform = "0x1337000000000000000000000000000000000000000000000000000000000000";
+    const checksum = "0x1111110000000000000000000000000000000000000000000000000000000000";
+
+    // only the owner of the client can add a release
+    try {
+      await operations.addChecksum(
+        release,
+        platform,
+        checksum,
+        { from: accounts[1] },
+      );
+    } catch(error) {
+      assert(error.message.includes("revert"));
+    }
+
+    let new_checksum = await operations.checksum("parity", release, platform);
+    assert.equal(new_checksum, 0);
+
+    // we successfully add a checksum for a release for
+    await operations.addChecksum(release, platform, checksum);
+
+    // the new checksum should be returned by the getter
+    new_checksum = await operations.checksum("parity", release, platform);
+    assert.equal(new_checksum, checksum);
+
+    // the checksum should map to the release and platform
+    const [new_release, new_platform] = await operations.build("parity", checksum);
+    assert.equal(new_release, release);
+    assert.equal(new_platform, platform);
+
+    // it should emit a `ChecksumAdded` event
+    const events = await watcher.get();
```

Comment:
We should add a test when `addChecksum` is called again with different parameters

Reply:
Fixed.